app/javascript/gamesFactory.js

```diff
@@ -189,16 +256,27 @@ angular.module('dappChess').factory('games', function (crypto, navigation, gameS
    */
   games.update = function (game) {
     console.log('update game called', game);
-    for(let i in games.list) {
-      if(games.list[i].gameId === game.gameId) {
-        games.list[i] = game;
+    for (let i in games.list) {
+      if (games.list[i].gameId === game.gameId) {
+        // update existing game
+        let g = games.list[i];
```

Comment:
I only used the for loop because I wasn't sure if arr.find would return a reference, but it does; so you can replace the whole loop by let g = games.getGame(game.gameId);